Library/Homebrew/rubocops/homepage_cop.rb

```diff
@@ -0,0 +1,78 @@
+require_relative "./extend/formula_cop"
+
+module RuboCop
+  module Cop
+    module FormulaAudit
+      # This cop audits `homepage` url in Formulae
+      class Homepage < FormulaCop
+        def audit_formula(_node, _class_node, _parent_class_node, formula_class_body_node)
+          homepage_node = find_node_method_by_name(formula_class_body_node, :homepage)
+          homepage = homepage_node.nil? ? "" : string_content(parameters(homepage_node).first)
+
+          problem "Formula should have a homepage." if homepage_node.nil? || homepage.empty?
+          problem "The homepage should start with http or https (URL is #{homepage})." unless homepage =~ %r{^https?://}
+
+          # Check for http:// GitHub homepage urls, https:// is preferred.
+          # Note: only check homepages that are repo pages, not *.github.com hosts
+          problem "Please use https:// for #{homepage}" if homepage.start_with? "http://github.com/"
```

Comment:
I wonder if it's worth turning this and everything below into a big `case` statement? What do you think?